src/Components/ServiceScene/Breakdowns/LabelBreakdownScene.tsx

```diff
@@ -36,74 +34,113 @@ import { getLabelOptions, sortLabelsByCardinality } from 'services/filters';
   });
 
-  constructor(state: Partial<LabelBreakdownSceneState>) {
+  // Labels/options can be passed in when instantiated, but should ONLY exist on the state of the variable
+  constructor(state: Partial<LabelBreakdownSceneState> & { options?: VariableValueOption[]; value?: string }) {
     super({
       ...state,
       $variables:
         state.$variables ??
         new SceneVariableSet({
-          variables: [new CustomVariable({ name: VAR_LABEL_GROUP_BY, defaultToAll: true, includeAll: true })],
+          variables: [
+            new CustomConstantVariable({
+              name: VAR_LABEL_GROUP_BY,
+              defaultToAll: false,
+              includeAll: true,
+
+              value: state.value ?? ALL_VARIABLE_VALUE,
+              options: state.options ?? [],
+            }),
+          ],
         }),
-      labels: state.labels ?? [],
       loading: true,
       sort: new SortByScene({ target: 'labels' }),
       search: new BreakdownSearchScene(),
+      value: state.value,
     });
 
     this.addActivationHandler(this.onActivate.bind(this));
   }
 
   private onActivate() {
+    this.setState({
+      loading: true,
+    });
+
     this.subscribeToEvent(SortCriteriaChanged, this.handleSortByChange);
 
     const variable = this.getVariable();
+    const serviceScene = sceneGraph.getAncestor(this, ServiceScene);
 
-    variable.subscribeToState((newState, oldState) => {
-      if (
-        newState.options !== oldState.options ||
-        newState.value !== oldState.value ||
-        newState.loading !== oldState.loading
-      ) {
-        this.updateBody(variable);
-      }
-    });
+    // Need to update labels with current state
+    if (serviceScene.state.labels) {
+      this.updateLabels(serviceScene.state.labels);
+    }
 
-    this.updateBody(variable);
+    this._subs.add(serviceScene.subscribeToState(this.onServiceStateChange));
+    this._subs.add(variable.subscribeToState(this.onVariableStateChange));
   }
 
-  private getVariable(): CustomVariable {
+  /**
+   * Update body when variable state is updated
+   * @param newState
+   * @param oldState
+   */
+  private onVariableStateChange = (newState: CustomConstantVariableState, oldState: CustomConstantVariableState) => {
+    if (
+      JSON.stringify(newState.options) !== JSON.stringify(oldState.options) ||
+      newState.value !== oldState.value ||
+      newState.loading !== oldState.loading
+    ) {
+      const variable = this.getVariable();
+      this.updateBody(variable, newState);
+    }
+  };
+
+  /**
+   * Pull the detected_labels from our service scene, update the variable when they change
+   * @param newState
+   * @param prevState
+   */
+  private onServiceStateChange = (newState: ServiceSceneState, prevState: ServiceSceneState) => {
+    const variable = this.getVariable();
+    if (JSON.stringify(newState.labels) !== JSON.stringify(prevState.labels)) {
```

Comment:
It would be so nice and elegant if we were able to trust these arrays and simply compare references, without having to stringify...

Reply:
Yeah a helper function is the only thing I can think of that might help, but that won't help if people don't remember that ```['a'] === ['a'] === false``` and call it in the first place.

Reply:
Actually if we had a way to throw a linting error when comparing arrays/objects in the state subscription callback? Although that could get messy too

Reply:
Yeah, dw. I was mainly thinking out loud. I don't think this would be possible in this version of the app.

Reply:
I created an issue to deal with this more generically: https://github.com/grafana/explore-logs/issues/614.